Feature #1247 automated testing

maartenhammer opened the pull request:
Closes #1247

maartenhammer committed:
#1247 Initial push of testing script

maartenhammer committed:
#1247 Initial push of testing script

maartenhammer committed:
#1247 Added Compile step, fixed summarize step

maartenhammer committed:
#1247 Added Compile step, fixed summarize step

maartenhammer committed:
#1247 added Version summary and pulling/resetting flavor and dataset to clean

maartenhammer committed:
#1247 added Version summary and pulling/resetting flavor and dataset to clean

maartenhammer committed:
#1247 Removed paralels for now

maartenhammer committed:
#1247 Removed paralels for now

maartenhammer committed:
#1247 Cleanup

maartenhammer committed:
#1247 Cleanup

maartenhammer committed:
#1247 added niceness variable

maartenhammer committed:
#1247 added niceness variable

maartenhammer committed:
#1247 Fixed VersionFile, Fixed SPMTest issue

maartenhammer committed:
#1247 Fixed VersionFile, Fixed SPMTest issue

maartenhammer committed:
#1247 removed unneccesary ` from code

maartenhammer committed:
#1247 removed unneccesary ` from code

maartenhammer committed:
#1247 added initialization to TestDataSet

maartenhammer committed:
#1247 added initialization to TestDataSet

maartenhammer committed:
#1247 Prevents crashing if comparisonTable empty

maartenhammer committed:
#1247 Prevents crashing if comparisonTable empty

maartenhammer committed:
#1247 Removed cleansing of TestDataSet

maartenhammer committed:
Removed cleansing of TestDataSet

maartenhammer committed:
#1247 Changed Spm_Json reading to new version

maartenhammer committed:
Changed Spm_Json reading to new version

maartenhammer committed:
removed flavor calls

maartenhammer committed:
#1247 removed flavor calls

maartenhammer committed:
Merge branch 'develop' into feature-#1247-Automated_Testing

maartenhammer committed:
#1247 added more try-catch

maartenhammer committed:
#1247 added more try-catch

maartenhammer committed:
Merge branch 'develop' into feature-#1247-Automated_Testing

maartenhammer committed:
Merge branch 'develop' into feature-#1247-Automated_Testing

maartenhammer committed:
Merge branch 'develop' into feature-#1247-Automated_Testing

maartenhammer committed:
#1247 changed standard locations to different dir

maartenhammer committed:
changed standard locations to different dir

maartenhammer committed:
Merge branch 'develop' into feature-#1247-Automated_Testing

maartenhammer committed:
#1247 new loaction in automated test script

maartenhammer committed:
#1247 new loaction in automated test script

maartenhammer committed:
Merge branch 'feature-#1247-Automated_Testing' of https://github.com/ExploreASL/ExploreASL into feature-#1247-Automated_Testing

jan-petr commented on Testing/xASL_test_SPM.m:
```diff
@@ -0,0 +1,126 @@
+function xASL_test_SPM(TestDirDest, bTestDataUsed)
```
I'm confused. This function used to be inside xASL_test_TestExploreASL. I'm fine having it independently, but then we should remove it from the other function to avoid having two versions.

maartenhammer replied:
I'll remove the one in xASL_test_TestExploreASL.

jan-petr commented on Testing/xASL_test_CompareReference.m:
```diff
@@ -0,0 +1,223 @@
+function [result] = xASL_test_CompareReference(pathReference, pathResults)
```
Is this copy pasted without changes from xASL_test_ExploreASL - then these should all be removed from there to avoid having two copies of the same code.

maartenhammer replied:
Some changes were made, but I'll change xASL_test_ExploreASL to call this instead of something else.

jan-petr commented on Testing/xASL_test_Summarize.m:
```diff
@@ -0,0 +1,110 @@
+function [summary, summaryTable] = xASL_test_Summarize(ResultsDir)
```
A new function or copied from elsewhere?

maartenhammer replied:
This is a new Function.

jan-petr reviewed (changes requested):
See the individual comments.
Also - I see many merge commits, were there any issues with conflicts? I guess not because you didn't really change the testing functions itself, right? 
For merging, we should probably squash-merge this or cherry-pick to another branch to get rid of the distracting merge commits.

maartenhammer committed:
#1247 removed redundant test_SPM

maartenhammer committed:
#1247 removed redundant compare tables

maartenhammer committed:
#1247 integrated Test_CompareReference in TestExploreASL

maartenhammer committed:
#1247 Moved WeeklyTest to CustomScripts

maartenhammer commented:
Removed all redundant files and moved the weeklyTest main script to CustomScripts with a Json-like input document. 
Currently running xASL_test_TestExploreASL to ensure changes didn't break the test script itself.

jan-petr reviewed (approved):
OK.

maartenhammer committed:
#1247 xASL_test_TestExploreASL bug in DifferenceTable fixed

maartenhammer committed:
Merge branch 'develop' into feature-#1247-Automated_Testing
